push_pattern_test.go

```diff
@@ -1,361 +0,0 @@
-package main
-
-import (
-	"testing"
-
-	"github.com/phrase/phraseapp-client/internal/paths"
-)
-
-type Patterns []*Pattern
-
-type Pattern struct {
-	File         string
-	Ext          string
-	TestPath     string
-	ExpectedCode string
-	ExpectedName string
-	ExpectedTag  string
-}
-
-func TestSpecialCharacters(t *testing.T) {
-	patterns := Patterns{
-		{
-			File:     "./locales/?*.yml",
-			Ext:      "yml",
-			TestPath: "locales/?en.yml",
-		},
-		{
-			File:         "./abc+/defg./}{x/][etc??/<locale_code>.yml",
-			Ext:          "yml",
-			TestPath:     "abc+/defg./}{x/][etc??/en.yml",
-			ExpectedCode: "en",
-		},
-	}
-	patterns.TestPatterns(t)
-}
-
-func TestSinglePlaceholders(t *testing.T) {
-	patterns := Patterns{
-		{
-			File:         "./.abc/<locale_code>.yml",
-			Ext:          "yml",
-			TestPath:     ".abc/en.yml",
-			ExpectedCode: "en",
-		},
-		{
-			File:     "./*.yml",
-			Ext:      "yml",
-			TestPath: "en.yml",
-		},
-		{
-			File:     "./locales/en.yml",
-			Ext:      "yml",
-			TestPath: "locales/en.yml",
-		},
-		{
-			File:     "./locales/.yml",
-			Ext:      "yml",
-			TestPath: "locales/en.yml",
-		},
-		{
-			File:         "./abc/defg/<locale_code>.lproj/.strings",
-			Ext:          "strings",
-			TestPath:     "abc/defg/en.lproj/Localizable.strings",
-			ExpectedCode: "en",
-		},
-		{
-			File:         "./locales/<locale_code>.yml",
-			Ext:          "yml",
-			TestPath:     "locales/en.yml",
-			ExpectedCode: "en",
-		},
-		{
-			File:         "./abc/<locale_code>.lproj/.strings",
-			Ext:          "strings",
-			TestPath:     "abc/en.lproj/Localizable.strings",
-			ExpectedCode: "en",
-		},
-	}
-	patterns.TestPatterns(t)
-}
-
-func TestMultiplePlaceholders(t *testing.T) {
```

Comment:
In the new version, globbing and placeholder resolving is done separately. Because of this, the `Resolve` method does not support the `**` operator anymore. Tests including that operator and placeholders to resolve couldn't be moved to either testing method directly. Those that could be moved to `TestResolve()` can be found [here](https://github.com/phrase/phraseapp-client/blob/master/internal/placeholders/placeholders_test.go), tests equivalent to the ones in here testing the `**` operator are now [here](https://github.com/phrase/phraseapp-client/blob/master/internal/paths/glob_test.go).
Note that support of the `**` operator in config files did not change. Before resolving placeholders, paths are split at the `**` operator and placeholders are resolved in the two halves independently. This makes the placeholder resolution code simpler. See https://github.com/phrase/phraseapp-client/pull/96/files#diff-1791d7629052d6b84cf1dde0701cc3f1R25.